geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -693,15 +693,19 @@ public boolean put(Object object) throws InterruptedException, CacheException {
 
     boolean isDREvent = isDREvent(sender.getCache(), value);
 
-    Region region = value.getRegion();
-    String regionPath = null;
-    if (isDREvent) {
-      regionPath = region.getFullPath();
-    } else {
+    String regionPath = value.getRegionPath();
+    if (!isDREvent) {
+      Region region = sender.getCache().getRegion(regionPath);
```

Comment:
sender.getCache().getRegion(regionPath) should not be null due to initialization as it invokes region.waitOnInitialization(). If region is null, I think it indicates region is destroyed.

Reply:
I will change to sender.getCache().getRegion(regionPath,true). 

It will be null. In the test, the region is destroyed. But we still need to get the root region. The event will be ignored later by if (!this.userRegionNameToShadowPRMap.containsKey(regionPath)) {